proposals/NNNN-package-access-modifier.md

```diff
@@ -0,0 +1,226 @@
+* Authors: [Ellie Shin](https://github.com/elsh), [Alexis Laferriere](https://github.com/xymus)
+* Review Manager: [John McCall](https://github.com/rjmccall)
+* Status: **Awaiting review**
+* Implementation: [apple/swift#61546](https://github.com/apple/swift/pull/62700), [apple/swift#62704](https://github.com/apple/swift/pull/62704), [apple/swift#62652](https://github.com/apple/swift/pull/62652)
+* Review: ([pitch](https://forums.swift.org/t/new-access-modifier-package/61459))
+
+## Introduction
+
+This proposal introduces `package` as a new access modifier. Currently, to access a symbol in another module, that symbol needs to be declared `public`. However, a symbol being `public` allows it to be accessed from any module at all, both within a package and from outside of a package, which is sometimes undesirable. We need a new access modifier to enable a more fine control over the visibility scope of such symbols. 
+
+## Motivation
+
+At the most basic level, every Swift program is just a collection of declarations: functions, types, variables, and so on. In principle, every level of organization above this is arbitrary; all of those declarations could be piled into a single file, compiled, and run. In reality, Swift programs are organized into separate files, directories, libraries, and so on. At each level, this organization reflects programmer judgment about relationships, both in the code and in how it is developed.
+
+As a language, Swift recognizes some of these levels. Modules are the smallest unit of library structure, with an independent interface and non-cyclic dependencies, and it makes sense for Swift to recognize that in both namespacing and access control. Files are the smallest grouping beneath that and are often used to collect tightly-related declarations, so they also make sense to respect in access control.
+
+Packages, as expressed by the Swift Package Manager, are a unit of code distribution. Some packages contain just a single module, but it's frequently useful to split a package's code into multiple modules. For example, when a module contains some internal helper APIs, those APIs can be split out into a utility module and maybe reused by other modules or packages.
+
+However, because Swift does not recognize organizations of code above the module level, it is not possible to create APIs like this that are purely internal to the package. To be usable from other modules within the package, the API must be public, but this means it can also be used outside of the package. This allows clients to form unwanted source dependencies on the API. It also means the built module has to export the API, which has negative implications for code size and performance.
+
+For example, here’s a scenario where a client (App) has access to a utility API from a package (gamePkg) it depends on.
+
+```
+App (Xcode project or appPkg)
+  |— Game (gamePkg)
+      |— Engine (gamePkg)
+```
+
+Here are source code examples.
```

Comment:
Putting this sentence makes it feel like you're introducing new examples.  What you're actually doing is continuing the scenario you just started.  The best way to express this, I think, is to skip the little tree above, because honestly the dependency is so simple that you're distracting from the example.  Just describe the dependency in the prose that starts the example, then dive into the code.